include/boost/config/auto_link.hpp

```diff
@@ -184,7 +187,16 @@ BOOST_LIB_VERSION:    The Boost version, in the form x_y, for Boost version x.y.
      // vc14.2:
 #    define BOOST_LIB_TOOLSET "vc142"
 
-#  elif defined(__BORLANDC__)
+#  elif defined(BOOST_EMBTC_WINDOWS)
+
+     // Embarcadero Clang based compilers:
+#    if defined(BOOST_EMBTC_WIN32C)
+#      define BOOST_LIB_TOOLSET "bcb32"
+#    elif defined(BOOST_EMBTC_WIN64)
+#      define BOOST_LIB_TOOLSET "bcb64"
+#    endif
```

Comment:
I'm changing this logic to simply `bcb` in #346. This makes it easier to match the toolset name used by Boost.Build (in the "versioned" layout).

Reply:
The toolset used by Boost.Build to build Boost for the Embarcadero C++ clang-based compilers should be 'embarcadero.jam' and not 'borland.jam'.

I think it is very bad if building for the old bcc32 compiler ( although it will fail miserably with current Boost ) should create the exact same library names as building for the clang-based compilers. That is why I chose different names for the Embarcadero C++ clang-based compilers. However I think you are right that there should not be different toolsets based on 32bit or 64bit builds, as no other compiler does that, so I would choose 'embtc' or maybe just 'emb', for the Embarcadero C++ clang-based compilers rather than the bcb32/bcb64 above. But I would not choose the same toolset name as for the old bcc32 compiler, as it will be confusing to the Embarcadero end-user, who may be used to seeing that name refer to the old compiler.

Reply:
There is nowadays (since v10.3.3) a linker warning against this problem: https://blogs.embarcadero.com/issues-mixing-classic-and-clang-objects-linker-warnings-improving-app-stability/

Reply:
In Embarcadero's home-grown version of Boost 1.68/1.70 their lines are:

// Embarcadero Clang based compilers:
#    if defined(BOOST_EMBTC_WIN32C)
#      define BOOST_LIB_TOOLSET "bcb32c"
#    elif defined(BOOST_EMBTC_WIN64)
#      define BOOST_LIB_TOOLSET "bcb64"
#    endif

I changed this slightly to "bcb32" from "bcb32c", since the Embarcadero C++ clang-based compilers are bcb32c and bcb32x, with the exact same clang-based code in all situations, and the only difference being that bcb32c uses the old bcc32 compiler options while bcb32x uses clang-based compiler options. They created separate names for their Embarcadero C++ clang-based BOOST_LIB_TOOLSET and their old bcc32 BOOST_LIB_TOOLSET. I think that separation should continue. But I agree with you that Boost usually does not create separate BOOST_LIB_TOOLSET names for 32bit and 64bit builds. So I still think that giving a BOOST_LIB_TOOLSET name that is different from the 'bcb' name is the way to go. So I would choose 'embtc', which is the general mnemonic Embarcadero used in their home-grown version of 1.68/1.70 to indicate the clang-based compilers.